services/virtualmachine/sdk_env.go

```diff
@@ -97,18 +98,21 @@ func (s *service) handleSdkEnvGetBlockProposerAddress(executionContext *executio
 
 // outputArg0: value array of (bytes)
 func (s *service) handleSdkEnvGetBlockCommittee(ctx context.Context, executionContext *executionContext, args []*protocol.Argument) ([][]byte, error) {
-	// TODO POSV2 should get ref as input ?
 	if len(args) != 0 {
 		return [][]byte{}, errors.Errorf("invalid SDK env getBlockProposerAddress args: %v", args)
 	}
 
-	// TODO POSV2 should be seperate provider ?
 	var committeeNodeAddresses []primitives.NodeAddress
 	var err error
 	committeeNodeAddresses, err = s.callElectionsSystemContract(ctx, executionContext)
 
 	if err != nil || len(committeeNodeAddresses) == 0 {
-		committeeNodeAddresses = s.committeeProvider.GetCommittee(ctx, uint64(executionContext.currentBlockHeight))
+		output, err2 := s.management.GetCommittee(ctx, &services.GetCommitteeInput{Reference: executionContext.lastBlockReferenceTime})
+		if err2 == nil {
+			committeeNodeAddresses = output.Members
+		} else {
```

Comment:
consider keeping only `if err2 != nil { s.logger...; return [][]byte{}, nil}` and then you don't need the else here...

Reply:
done